components/ipfs/browser/ipfs_tab_helper.cc

```diff
@@ -30,11 +28,17 @@ IPFSTabHelper::IPFSTabHelper(content::WebContents* web_contents)
 
 // static
 void IPFSTabHelper::MaybeCreateForWebContents(
-    content::WebContents* web_contents) {
-  if (!ipfs::IpfsService::IsIpfsEnabled(web_contents->GetBrowserContext()))
+    content::WebContents* web_contents,
+    bool regular_profile,
+    IpfsTabHelperDelegate* delegate) {
+  auto* browser_context = web_contents->GetBrowserContext();
+  if (!ipfs::IpfsService::IsIpfsEnabled(browser_context, regular_profile)) {
```

Comment:
I'm a bit concerned that this can potentially be out of sync with `IpfsServiceFactory::GetForContext`. I think it would be better to pass in the IpfsService and check for `nullptr`